### DISCUSSION TITLE
HomeAssistantError: value should be a string for dictionary value @ data['entity_id']

### DISCUSSION THREAD

**Discussion by HonestJohn61** [zachowj/node-red-contrib-home-assistant-websocket#1203] [Q&A]
Hi, I don't understand this error message. As far as I understand it, it's saying that HA is expecting a string value for 'entity_id', but that's exactly what it's getting as far as I can see in the simple test I've set up. What am I missing? I'd really appreciate a hand here. Thanks

<img width="1428" alt="image" src="https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/29650722/7b7ed96c-3089-4a23-a1d7-feae0a48fd79">

**Comment by zachowj** [comment]
That's an error directly from Home Assistant. Not exactly sure what it thinks it bad. Have you tried the same service call in the developer tools in HA?

**Reply by HonestJohn61** [reply]
Thanks for the reply. I have just tried the service call in the developer tools in HA. The format of the call needs changing slightly from the Node-RED debug report. Specifically the domain needs combining with the service name, and the entity_id field needs to appear under data not target, but I guess this reformatting is a standard part of the integration? So this works:

<img width="1167" alt="image" src="https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/29650722/a7c06a1a-946a-4cb2-b535-1aea44e74a92">

The errors reported in the developer tools I encountered getting there were these:

<img width="1167" alt="image" src="https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/29650722/fac672dc-1c8d-47fe-9e82-9d4a028ce863">

<img width="1167" alt="image" src="https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/29650722/aaa1ac23-e0a3-4549-8f9c-a0893d60e715">

Does this help at all?

**Comment by HonestJohn61** [comment]
Having done that, I now have a workaround, but it would be good to understand where the error lies and eliminate it, of course.

<img width="1218" alt="image" src="https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/29650722/fbdb3045-53b2-4528-9675-654fa9cef6b1">

<img width="1126" alt="image" src="https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/29650722/3775c25f-46f4-4cde-870c-bcb1891c20d5">

**Comment by HonestJohn61** [comment]
For information, this workaround also works for this other evohome service which otherwise fails in the same way:

<img width="1220" alt="image" src="https://github.com/zachowj/node-red-contrib-home-assistant-websocket/assets/29650722/875bbcee-c451-4ef1-a40e-4671652d51b2">
